gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -454,18 +520,51 @@ public FileStore getFileStore(Path path) {
   public boolean equals(@Nullable Object other) {
     return this == other
         || other instanceof CloudStorageFileSystemProvider
-        && Objects.equals(gcsService, ((CloudStorageFileSystemProvider) other).gcsService);
+        && Objects.equals(storage, ((CloudStorageFileSystemProvider) other).storage);
```

Comment:
You are aware that `Storage` does not override the `equals` method...
I think a better check would be to compare their options (`StorageOptions`).